### DISCUSSION TITLE
Sobre el DevConteiner mencionado en el enunciado

### DISCUSSION THREAD

**Discussion by MatiLaB** [IIC2133-PUC/2024-2#101] [Tarea 1] [answered]
Me entró la duda con la mención del `.devcontainer` en el enunciado. ¿Cómo nos podemos asegurar que este esté implementado correctamente? Para la T0 no recuerdo que se haya pedido y nunca me preocupe de eso. ¿Ya está implementado con el código que se encuentra en el `devcontainer.json`? ¿O tenemos que editarlo según el link que se encuentra ahí?

Ahora me he dado cuenta de que al abrir vscode me sale si quiero reabrir el repositorio en el ambiente del `devcontainer` pero al hacerlo me sale un error de que no está corriendo. 

En resumen, quería saber si es algo que tenemos que revisar nosotros o ya viene dado. En todo caso también me gustaría saber como saber si el programa cumple con lo pedido dentro de este ambiente.

**Comment by tomastrivino** [comment, marked as answer]
Hola,
El enunciado refiere a que tu tarea debería ejecutarse sin problemas en el entorno `devcontainer` que se te entrega (para más información sobre los mismos, véase [este enlace](https://code.visualstudio.com/docs/devcontainers/create-dev-container)). No debes de editar el archivo `json`. La guía para el setup la puedes encontrar [haciendo click acá](https://github.com/IIC2133-PUC/edd-dev-container?tab=readme-ov-file#edd-devcontainer). 

En resumen,
- La compilación puede ser distinta dependiendo del sistema operativo / entorno en el cual se trabaje (en tu computador podría compilar adecuadamente; en el mío no, por darte un ejemplo).
- Para estandarizar la corrección, se utiliza el entorno definido en el devcontainer que se te entrega.
- Por ende, si tu tarea se ejecuta adecuadamente en el mismo, te asegurarás de que se pueda compilar al momento en que se te haga la corrección.

**Reply by tomastrivino** [reply]
¡ Y asegúrate de tener Docker instalado !

**Reply by MatiLaB** [reply]
Y después de instalar eso e reabrir con el `devcontainer`, como puedo probar el código? Que logré encontrar una terminal para hacerlo pero no me reconoce los archivos incluso después de correr `make`.

Update: si me funciona correrlo en el `devcontainer `(logré encontrar el nuevo directorio dentro de docker) pero al intentar correrlo el programa principal, uno que ya me funcioba antes, sale que hay un segmentation fault. Revisando me fije que en el que había hecho y el que aún no he hecho mucho sale el mismo error con cerrar el archivo del output o al hacer `fprintf` en él. Me di cuenta probando que al hacerlo con `print` si corre, pero igual sale error al final. Esto no debería pasar o si? Será por la forma que docker maneja los archivos y que no los puede editar? (intenté eso último y no me dejaba así que si el problema es la escritura en el archivo puede ser eso) Será necesario que pasé todos los `printf` a `fprintf` para probarlo en el docker o habrá otra forma? Porque fuera de eso si corren.

**Reply by tomastrivino** [reply]
Asegúrate de que el input que pones exista

**Reply by MatiLaB** [reply]
Pero por ejemplo, al poner `./order tests/order/easy/input.txt output.txt` luego de reabrir la carpeta con el dev container me sale un segmentation fault que no me salía antes. Revisando con `valgrind` me di ucenta de que tiene que ver con `frpintf`. Porque lo probé con la primera parte y si lo dejo como `printf` solamente si lo retorna, pero al cerrar el archivo tiene problemas y retorna el segmentation fault. Me sale lo siguiente al probarlo:

![image](https://github.com/user-attachments/assets/4718d7da-5423-47ef-b380-b9464c822d2f)

¿Dices que esto debe ser un simple problema del input que estoy dando o podría ser algo del ambiente que no permite que corra bien si el programa no cumple con lo pedido? Quería saber bien para asegurarme como es que se prueba en el dev y que no me perjudique después.

**Reply by tomastrivino** [reply]
**Actualización: el error se debía a un tema de permisos en el devcontainer. Si presentan un error similar, intenten ejecutando el comando `sudo chmod 777 -R *`, el que modificará los permisos dentro del contenedor.**
